### ISSUE TITLE
where is the fine-tune config file ? 

### ISSUE THREAD

**Issue by Edwardmark** [microsoft/GLIP#5]
hi，could you please tell me where the fine-tune config file is? 
I want to do prompt tuning. Thanks.

**Comment by liunian-harold-li** [comment]
Hi the commands for running the prompt tuning is at https://github.com/microsoft/GLIP#prompt-tuning.

**Comment by Edwardmark** [comment]
> Hi the commands for running the prompt tuning is at https://github.com/microsoft/GLIP#prompt-tuning.

![image](https://user-images.githubusercontent.com/15247551/166854333-3f99fd80-938e-4ed9-b069-f34ccc0f3a6f.png)
which file is the configs for ft-task? Thanks.

**Comment by Edwardmark** [comment]
@liunian-harold-li could you use Aquarium_Aquarium_Combined.v2-raw-1024.coco.yaml data as an example to show how to do prompt / full fine-tuning, and give the full command with all config and options? It would make things more clear. Thanks.

**Comment by liunian-harold-li** [comment]
Sorry for the late reply. The {configs} is configs/odinw/Aquarium_Aquarium_Combined.v2-raw-1024.coco.yaml. The explanation about this config is given at https://github.com/microsoft/GLIP#odinw--custom-dataset-evaluation. The fine-tuning config is the same as the evaluation config. Sorry for the confusion.

**Comment by Edwardmark** [comment]
@liunian-harold-li Thanks. So I should set ft-config {configs} as " configs/odinw/Aquarium_Aquarium_Combined.v2-raw-1024.coco.yaml" the same as --config-file {config_file}? Is that right?

**Comment by liunian-harold-li** [comment]
Sorry for the confusion. --config-file should be set to the pretrian
model's config file. An example command:

python -m torch.distributed.launch --nproc_per_node=4 tools/finetune.py \
      --config-file configs/pretrain/glip_Swin_T_O365_GoldG.yaml
--ft-tasks configs/odinw/Aquarium_Aquarium_Combined.v2-raw-1024.coco.yaml
--skip-test \
      --custom_shot_and_epoch_and_general_copy 3_200_4 \
      --evaluate_only_best_on_test --push_both_val_and_test \
      MODEL.WEIGHT MODEL/glip_tiny_model_o365_goldg_cc_sbu.pth \
      SOLVER.USE_AMP True TEST.DURING_TRAINING True TEST.IMS_PER_BATCH
4 SOLVER.IMS_PER_BATCH 4 SOLVER.WEIGHT_DECAY 0.05 TEST.EVAL_TASK
detection DATASETS.TRAIN_DATASETNAME_SUFFIX _grounding
MODEL.BACKBONE.FREEZE_CONV_BODY_AT 2 MODEL.DYHEAD.USE_CHECKPOINT True
SOLVER.FIND_UNUSED_PARAMETERS False SOLVER.TEST_WITH_INFERENCE True
SOLVER.USE_AUTOSTEP True DATASETS.USE_OVERRIDE_CATEGORY True
SOLVER.SEED 10 DATASETS.SHUFFLE_SEED 3 DATASETS.USE_CAPTION_PROMPT
True DATASETS.DISABLE_SHUFFLE True \
      SOLVER.STEP_PATIENCE 3 SOLVER.CHECKPOINT_PER_EPOCH 1.0
SOLVER.AUTO_TERMINATE_PATIENCE 8 SOLVER.MODEL_EMA 0.0
SOLVER.TUNING_HIGHLEVEL_OVERRIDE full

**Comment by Edwardmark** [comment]
@liunian-harold-li Thank you very much, your kind reply is really helpful.

**Comment by Edwardmark** [comment]
@liunian-harold-li Hi, I try to run the finetune.py with the following command:
```
CUDA_VISIBLE_DEVICES=0 python tools/finetune.py \
        --config-file configs/pretrain/glip_Swin_L.yaml \
        --ft-tasks configs/odinw/Aquarium_Aquarium_Combined.v2-raw-1024.coco.yaml \
        --skip-test \
        --custom_shot_and_epoch_and_general_copy 0_200_1 \
        --evaluate_only_best_on_test --push_both_val_and_test \
        MODEL.WEIGHT MODEL/glip_large_model.pth \
        SOLVER.USE_AMP True \
        TEST.DURING_TRAINING True \
        TEST.IMS_PER_BATCH 1 \
        SOLVER.IMS_PER_BATCH 1 \
        TEST.EVAL_TASK detection \
        DATASETS.TRAIN_DATASETNAME_SUFFIX _grounding \
        MODEL.BACKBONE.FREEZE_CONV_BODY_AT 2 \
        MODEL.DYHEAD.USE_CHECKPOINT True \
        SOLVER.FIND_UNUSED_PARAMETERS False \
        SOLVER.TEST_WITH_INFERENCE True \
        SOLVER.USE_AUTOSTEP True \
        DATASETS.USE_OVERRIDE_CATEGORY True \
        SOLVER.SEED 10 \
        DATASETS.SHUFFLE_SEED 3 \
        DATASETS.USE_CAPTION_PROMPT True \
        DATASETS.DISABLE_SHUFFLE True \
        SOLVER.STEP_PATIENCE 2 \
        SOLVER.CHECKPOINT_PER_EPOCH 1.0 \
        SOLVER.AUTO_TERMINATE_PATIENCE 4 \
        SOLVER.MODEL_EMA 0.0 \
        SOLVER.WEIGHT_DECAY 0.05 \
        SOLVER.TUNING_HIGHLEVEL_OVERRIDE full
```
it shows the following error:
![image](https://user-images.githubusercontent.com/15247551/167101299-51b5f3be-a42b-423f-9413-750a01862070.png)

Then I debug by run the cmd using pdb, and I found that the error occurs in https://github.com/microsoft/GLIP/blob/main/maskrcnn_benchmark/data/build.py#L443, for dataset does not have attribute datasets, as follows:
![image](https://user-images.githubusercontent.com/15247551/167101929-1b70e72e-a001-4335-b322-67d94c15b92d.png)

I am wondering,if the code has some bugs, but I cannot find out. 

Could you please give a specific command in readme, and check whether this bug is reproduceble.  Thanks.

**Comment by liunian-harold-li** [comment]
Sorry for the confusion. Indeed, DATALOADER.DISTRIBUTE_CHUNK_AMONG_NODE should be set to false. 

It was intended to be used when we have a lot of pre-training data (dozens of terabytes) and one training node cannot load all the data. We implemented this to distribute the data to different nodes to avoid running out of hard disk space.

I have removed the option in the GLIP-L config.

**Comment by Edwardmark** [comment]
@liunian-harold-li Thank you very much.

**Comment by imaginistLi** [comment]
Excuse me, I want to ask another question: Where to download the PREDEFINED_TEXT 'odinw/pothole/category_description.json' metioned in configs?
